docs/applications/containers/introduction-to-microservices-with-kubernetes-and-gitlab/index.md

```diff
@@ -0,0 +1,210 @@
+gitlab-redis-6cc5f876db-hd44h                          0/2     PodInitializing   0          117s
+gitlab-registry-7f95847fb4-2x5bp                       1/1     Running           0          117s
+gitlab-registry-7f95847fb4-7cpg6                       1/1     Running           0          117s
+gitlab-sidekiq-all-in-1-74655c6c75-vcwp6               0/1     Init:2/3          0          119s
+gitlab-task-runner-68775bf44d-q27lh                    1/1     Running           0          119s
+gitlab-unicorn-694748c5c9-mb8kf                        0/2     Init:2/3          0          103s
+gitlab-unicorn-694748c5c9-w47tl                        0/2     Init:2/3          0          119s
+{{< /output >}}
+
+    The Helm chart will create a [NodeBalancer](/docs/platform/nodebalancer/getting-started-with-nodebalancers/), which will be used to add an additional layer of high availability to our application.
+
+1. Domain propagation will take a few minutes to successfully resolve to your new IP address following this change, and the `gitlab-runner` microservice may crash since it relies on this resolution. You'll want to periodically check this service to make sure it hasn't failed. See the [Troubleshooting section](/docs/applications/containers/introduction-to-microservices-with-kubernetes-and-gitlab/#troubleshooting) for tips on how to resolve Pods in statuses other than `Running`.
+
+        kubectl get pods | grep gitlab-runner
+
+1. Navigate to the [NodeBalancer](https://cloud.linode.com/nodebalancers) section of the Cloud Manager, and find the IP address of this new resource. The label of the NodeBalancer will be a randomly generated 32 character string, with your worker nodes pre-configured as Backend Linodes.
+
+1. If you have a domain set-up to point to one of your worker nodes as instructed above, as your containers come up you should see A records made automatically for each of the subdomains. They should point to your NodeBalancer. If not, you can create them manually.
+
+  -  minio.example.com
+  -  gitlab.example.com
+  -  registry.example.com
+
+1. Finally, once all services have initialized, you can navigate to `gitlab.example.com` using the domain name that you configured earlier, and after a few moments, you'll see a login screen for your new Microservices Application.
+
+![Gitlab Login](gitlablogin.png)
+
+## Troubleshooting
+
+### Gitlab-runner in CrashLoopbackOff Status
+
+If gitlab-runner is in the `CrashLoopbackOff` status, it has most likely timed out before your domains have resolved, and you'll need to delete the pod once domain propagation is completed for the service to restart and confirm your domain is working. You can do this by replacing the randomly generated string following `gitlab-gitlab-runner-` with the string that appeared in the previous command:
+
+    kubectl delete pods gitlab-gitlab-runner-dd97f84b-zplq9
+
+### Pods Stuck in Initialization Status
```

Comment:
I added this section because I needed to do this several times during testing and thought it would be helpful. Mostly I found networking issues related to the next section. I think left over from previous deployments that really should have been cleared out - and containers trying to connect with bad ip addresses. But this helped diagnose those issues.